compiler/rustc_mir_transform/src/coverage/mod.rs

```diff
@@ -204,6 +225,56 @@ fn inject_coverage_statements<'tcx>(
     }
 }
 
+/// For each conditions inject statements to update condition bitmap after it has been evaluated.
+/// For each decision inject statements to update test vector bitmap after it has been evaluated.
+fn inject_mcdc_statements<'tcx>(
+    mir_body: &mut mir::Body<'tcx>,
+    basic_coverage_blocks: &CoverageGraph,
+    coverage_spans: &CoverageSpans,
+) {
+    if coverage_spans.test_vector_bitmap_bytes() == 0 {
+        return;
+    }
+
+    // Inject test vector update first because inject statement always inject new statement at head.
+    for (end_bcbs, bitmap_idx) in
+        coverage_spans.all_bcb_mappings().filter_map(|mapping| match &mapping.kind {
+            BcbMappingKind::Decision { end_bcbs: end_bcb, bitmap_idx, .. } => {
+                Some((end_bcb, *bitmap_idx))
+            }
```

Comment:
`end_bcbs: end_bcb` looks like a mistake; you probably meant to change this to `end_bcbs`.